### PR TITLE
Separate built-in funs as primitives

### PR DESCRIPTION
**Pull request by tranma**
* Add a `PrimBuiltinFun` primitive to `Minimal`, which includes built-in maths functions, etc.
* Add `keys`, `vals` primitive to get the keys/vals out of the map
* Fix pretty printer to separate argument types with comma

@amosr @jystic

### CODE REVIEW

File: src/Icicle/Common/Exp/Prim/Minimal.hs [hunk 1]
```diff
@@ -47,7 +46,7 @@ data Prim
 data PrimArithUnary
  = PrimArithNegate
  | PrimArithAbsolute
- deriving (Eq, Ord, Show)
+ deriving (Eq, Ord, Show, Enum, Bounded)
```
Comment (amosr) [inline src/Icicle/Common/Exp/Prim/Minimal.hs]: what are these for?

Comment (tranma) [inline src/Icicle/Common/Exp/Prim/Minimal.hs]: to be used for `arbitraryBounedEnum`

**Comment by amosr** [comment]
:octopus:

File: src/Icicle/Common/Exp/Prim/Minimal.hs (hunk 1 above)
Comment (amosr) [inline src/Icicle/Common/Exp/Prim/Minimal.hs]: oh, nice

**Pull request merged**